1config-ui/src/com/brunobonacci/oneconfig/ui/core.cljs

```diff
@@ -0,0 +1,535 @@
+                      })))
+
+(defn get-footer-text [allState]
+  (GET "/footer"
+       {:handler          #(swap! allState assoc-in [:1config-version] %)
+        :format          :json
+        :response-format :json
+        :keywords?       true
+        :error-handler   error-handler}))
+
+(defn create-service-name-row [num sz key]
+  (if (zero? num) [:td {:row-span sz} key]))
+
+(defn create-table-extended [items]
+  (for [item items]
+    (let [{:keys [key env version change-num content-type master-key user]} item]
+      ^{:key (string/join "-" [key env version change-num content-type])}
+      [:tr {:class "top aligned"}
+       [create-service-name-row (.indexOf items item) (count items) key]
+       [:td {:data-label "Environment"} (comm/as-label (comm/colourize-label env) env)]
+       [:td {:data-label "Version"} version]
+       [:td {:data-label "Change num"} change-num]
+       [:td {:data-label "Time"} (comm/parse-date change-num)]
+       [:td {:data-label "Type"} (comm/as-label content-type)]
+       [:td {:data-label "Value"}
+        [:button {:class "ui icon button" :on-click #(get-config-item! item)}
+         [:i {:class "ellipsis horizontal icon"}]]
+        ]
+       [:td {:data-label "Master Key" :class "master-key-width"}
+        [:div {:class "tooltip"} (comm/get-kms-uuid master-key)
+         [:span {:class "tooltiptext"} master-key]]
+        ]
+       [:td {:data-label "User"}
+        [:div {:class "tooltip"} (comm/get-aws-username user)
+         [:span {:class "tooltiptext"} user]]
+        ]
+       ]
+      )
+    )
+  )
+
+(defn create-minified-table [items]
+  (for [item items]
+    (let [{:keys [key env version change-num content-type]} item]
+      ^{:key (string/join "-" [key env version change-num content-type])}
+      [:tr {:class "top aligned"}
+       [create-service-name-row (.indexOf items item) (count items) key]
+       [:td {:data-label "Environment"} (comm/as-label (comm/colourize-label env) env)]
+       [:td {:data-label "Version"} version]
+       [:td {:data-label "Change num"} change-num]
+       [:td {:data-label "Time"} (comm/parse-date change-num)]
+       [:td {:data-label "Type"} (comm/as-label content-type)]
+       [:td {:data-label "Value"}
+        [:button {:class "ui icon button" :on-click #(get-config-item! item)}
+         [:i {:class "ellipsis horizontal icon"}]]
+        ]
+       ]
+      )
+    )
+  )
+
+(defn notify-file-upload! [name]
+  (if (= name "")
+    (reset! file-upload-style "hide-element")
+    (reset! file-upload-style "show-element")))
+
+
+
+(defn sidenav-display!  []
+  (if (= @sidenav-display-toggle "sidenav visible")
+    (reset! sidenav-display-toggle "sidenav hidden")
+    (reset! sidenav-display-toggle "sidenav visible"))
+  )
+
+(defn debug-output!  []
+  (println "click event"))
+
+(defn add-config-entry-form [submitData]
+  [:form {:class "ui form" :on-submit #(add-config-entry! %1 (get @submitData :new-entry))}
```

Comment:
it is an anti-pattern to have multiple `deref` of the same atom in the same function.
the value could change in between two `deref` potentially leading to inconsistent data.
I'd recommend to pass the value to the `add-config-entry-form` function rather than the atom, like `(add-config-entry-form @submitData)`